src/test/java/com/hpe/adm/octane/ideplugins/integrationtests/IntegrationTestBase.java

```diff
@@ -13,36 +13,280 @@
+
+import java.lang.annotation.Annotation;
+import java.lang.annotation.Target;
+import java.lang.reflect.AnnotatedType;
+import java.util.Collection;
+
+import static org.junit.Assert.fail;
 
 /**
  * Enables the use of the {@link Inject} annotation
  */
 public abstract class IntegrationTestBase {
 
+
+    private final Logger logger = LogManager.getLogger(IntegrationTestBase.class.getName().toString());
+
     protected EntityGenerator entityGenerator;
 
+    ConnectionSettingsProvider connectionSettings;
+
+
+    /**
+     * This function will set up a context needed for the tests, the context is derived from the annotations set the
+     * implementing class
+     */
     @Before
-    public void setup () {
+    public void setup() {
+        Annotation[] annotations = this.getClass().getDeclaredAnnotations();
+
+        boolean cleanWorkspace = findWorkSpaceAnnotation(annotations);
+
+        if (cleanWorkspace) {
+            connectionSettings = PropertyUtil.readFormVmArgs() != null ? PropertyUtil.readFormVmArgs() : PropertyUtil.readFromPropFile();
+            ConnectionSettings cs = connectionSettings.getConnectionSettings();
+            cs.setWorkspaceId(createWorkSpace());
+            connectionSettings.setConnectionSettings(cs);
+        } else {
 
-        ConnectionSettingsProvider connectionSettings = PropertyUtil.readFormVmArgs() != null ?
-                PropertyUtil.readFormVmArgs() : PropertyUtil.readFromPropFile();
+            connectionSettings = PropertyUtil.readFormVmArgs() != null ? PropertyUtil.readFormVmArgs() : PropertyUtil.readFromPropFile();
+            ConnectionSettings cs = connectionSettings.getConnectionSettings();
+            long defaultWorkspaceId = getDefaultWorkspaceId();
 
-        if(connectionSettings == null){
+            if (defaultWorkspaceId > 0)
+                cs.setWorkspaceId(defaultWorkspaceId);
+            else {
+                cs.setWorkspaceId(createWorkSpace());
+            }
+            connectionSettings.setConnectionSettings(cs);
+        }
+
+        if (connectionSettings == null) {
             throw new RuntimeException("Cannot retrieve connection settings from either vm args or prop file, cannot run tests");
         }
 
+        boolean createNewUser = findUserAnnotation(annotations);
+
+        if(createNewUser) {
+            createUSer();
+        } else {
+            //find all users and choose the first one - if there are no users create one
+            //TODO
+        }
+
         Injector injector = Guice.createInjector(new ServiceModule(connectionSettings));
         injector.injectMembers(this);
         entityGenerator = new EntityGenerator(injector.getInstance(OctaneProvider.class));
     }
 
+    /**
+     * This method will look for the workspace annotation in the implementing subclass
+     * @param annotations - of the subclass
+     * @return boolean - found or not
+     */
+    private boolean findWorkSpaceAnnotation(Annotation[] annotations) {
```

Comment:
Can the find method be simplified into one?
Such as getAnnotation where you give it a param of Annotation type, as in Annotation.class, and the list of annotations.
It would return the Annotation instance if found, null otherwise. 
THe you have one method for both. Adittionaly I think it can be use generics for the return type. Not sure about this one